format.go

```diff
@@ -342,23 +351,46 @@ func (a *unknownAny) ProtoMessage() {
 
 var _ proto.Message = (*unknownAny)(nil)
 
-// RequestParserAndFormatterFor returns a request parser and formatter for the
+// FormatterOptions is a set of flags that are passed to a JSON or text formatter.
+type FormatterOptions struct {
+	EmitJSONDefaultFields bool // FormatJSON only
+	IncludeTextSeparator  bool // FormatText only
+	AllowUnknownFields    bool // FormatJSON only
+}
+
+// RequestParserAndFormatterWithOptions returns a request parser and formatter for the
 // given format. The given descriptor source may be used for parsing message
-// data (if needed by the format). The flags emitJSONDefaultFields and
-// includeTextSeparator are options for JSON and protobuf text formats,
-// respectively. Requests will be parsed from the given in.
-func RequestParserAndFormatterFor(format Format, descSource DescriptorSource, emitJSONDefaultFields, includeTextSeparator bool, in io.Reader) (RequestParser, Formatter, error) {
+// data (if needed by the format).
+// It accepts a set of options. The field EmitJSONDefaultFields and IncludeTextSeparator
+// are options for JSON and protobuf text formats, respectively. The AllowUnknownFields field
+// is a JSON-only format flag.
+// Requests will be parsed from the given in.
+func RequestParserAndFormatterWithOptions(format Format, descSource DescriptorSource, opts FormatterOptions, in io.Reader) (RequestParser, Formatter, error) {
```

Comment:
Let's shorten this to `RequestParserAndFormatter` and then deprecate the other version.

Also, please move `opts` to be the last argument since that is more idiomatic for functions that use an options-struct pattern.